pkg/controller/trainjob_controller.go

```diff
@@ -105,11 +107,29 @@ func (r *TrainJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 func (r *TrainJobReconciler) reconcileObjects(ctx context.Context, runtime jobruntimes.Runtime, trainJob *trainer.TrainJob) (objsOpState, error) {
 	log := ctrl.LoggerFrom(ctx)
 
-	objs, err := runtime.NewObjects(ctx, trainJob)
+	objects, err := runtime.NewObjects(ctx, trainJob)
 	if err != nil {
 		return buildFailed, err
 	}
-	for _, obj := range objs {
+	for _, object := range objects {
+		// TODO (astefanutti): Remove conversion to unstructured when the runtime.ApplyConfiguration
+		//  interface becomes available and first-class SSA method is added to the controller-runtime
+		// client. See https://github.com/kubernetes/kubernetes/pull/129313
+		var obj client.Object
+		if o, ok := object.(client.Object); ok {
+			return buildFailed, fmt.Errorf("unsupported type client.Object for component: %v", o)
+		} else {
+			u, err := apiruntime.DefaultUnstructuredConverter.ToUnstructured(object)
+			if err != nil {
+				return buildFailed, err
+			}
+			obj = &unstructured.Unstructured{Object: u}
```

Comment:
```suggestion
		}
		u, err := apiruntime.DefaultUnstructuredConverter.ToUnstructured(object)
		if err != nil {
			return buildFailed, err
		}
		obj = &unstructured.Unstructured{Object: u}
```

This still seems to remain.

Reply:
It should be good now.

Reply:
Thank you!